dapp/src/lib/utils/format/formatNames.ts

```diff
@@ -1,15 +1,29 @@
 // Copyright (c) 2025 IOTA Stiftung
 // SPDX-License-Identifier: Apache-2.0
 
-export function splitNameInParts(name: string) {
-    const parts = name.split('.');
+export function splitNameParts(nftName: string) {
+    const parts = normalizeNameInput(nftName).split('.');
+    const lastIndex = parts.length - 1;
+    const parentName = parts[lastIndex];
 
     return {
-        namePart: parts.slice(-2).join('.'),
-        subnamePart: parts.slice(0, -2).join('.'),
+        parentName,
+        subname: parts.slice(0, lastIndex).join('.'),
     };
 }
 
 export function normalizeNameInput(name: string) {
     return name.toLowerCase().replace(/\.iota$/i, '');
 }
+
+export function addNameSuffix(name: string) {
```

Comment:
Wdyt about (so it that its like the opposite of `normalizeNameInput`)
```suggestion
export function denormalizeNameInput(name: string) {
```